### ISSUE TITLE
stylesheet is broken as of 2.2

### ISSUE THREAD

**Issue by robot-ranger** [mtconnect/cppagent#378]
the stylesheet appears to be broken for me since 2.2. anyone else have this problem?

![image](https://github.com/mtconnect/cppagent/assets/93950632/e2901856-6d16-4269-9b25-75db3047256b)

When i change the agent.cfg and devices.xml back to `2.0`, it works as expected. `2.2` and it appears like above.

no issues with client-end apps; the xml appears to be in good shape, so i think its just the stylesheet

**Comment by wsobel** [comment]
This usually has to do with the namespace in the stylesheet. I'll check.

**Comment by robot-ranger** [comment]
> This usually has to do with the namespace in the stylesheet. I'll check.

I was trying my best to act like i know stylesheets and dove straight into the transform logic haha. looks like youre right!

![image](https://github.com/mtconnect/cppagent/assets/93950632/5de293a4-452b-4ee0-9f2e-f8dac9648d0d)


is it possible to make it read the schema version from the probe?

**Comment by robot-ranger** [comment]
or maybe it could be an env var? i dont know much about stylesheets, so i dont know how dynamic it could be or how it would work. but it would be nice to be able to roll up and down schema versions without editing the file

**Comment by robot-ranger** [comment]
oops - accidentally close it

**Comment by wsobel** [comment]
They are static documents, so we would need the agent to preprocess the file. Currently it is just serving it as a text file with caching for files under a certain size.  I can flag certain files for preprocessing. That wouldn’t be too difficult if it is a simple replace of tokens in the config file. On Dec 12, 2023, at 19:22, robot-ranger ***@***.***> wrote:﻿
oops - accidentally close it

**Comment by robot-ranger** [comment]
fixed in #403

**Comment by robot-ranger** [comment]
fyi - if anyone needs to fix this on an existing container, the following command works:

```
docker exec -it agent sed -i 's/2.0/2.2/g' /mtconnect/data/styles/styles.xsl
```

(replace `agent` with your contianer name if not `agent`)
